runtime/compiler/runtime/MethodMetaData.c

```diff
@@ -224,16 +224,16 @@ static VMINLINE TR_StackMapTable * initializeMapTable(J9JavaVM * javaVM, J9TR_Me
    return mapTable;
    }
 
-static VMINLINE TR_StackMapTable * findOrCreateMapTable(J9VMThread * currentThread, J9TR_MethodMetaData * metaData, UDATA fourByteOffsets)
+static VMINLINE TR_StackMapTable * findOrCreateMapTable(J9VMThread * walkThread, J9TR_MethodMetaData * metaData, UDATA fourByteOffsets)
    {
    TR_StackMapTable * mapTablePtr = 0;
    assert(metaData);
 
    // In a signal handler, do not use or create the map tables. The tables may be in an inconsistent
    // state when interrupted by the signal, and malloc must not be called from a signal handler.
-   if (J9_ARE_NO_BITS_SET(currentThread->privateFlags2, J9_PRIVATE_FLAGS2_ASYNC_GET_CALL_TRACE))
+   if (J9_ARE_NO_BITS_SET(walkThread->privateFlags2, J9_PRIVATE_FLAGS2_ASYNC_GET_CALL_TRACE))
```

Comment:
`walkThread` has no meaning here (this is not only called by the walker). Please pass the currentThread and javaVM, and change the bit test to either currentThread is NULL or the bit is set.